src/core/recv_buffer.c

```diff
@@ -76,6 +191,64 @@ QuicRecvChunkFree(
     }
 }
 
+//
+// Validate the receive buffer invariants.
+// No-op in release builds.
+//
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+QuicRecvBufferValidate(
+    _In_ const QUIC_RECV_BUFFER* RecvBuffer
+    )
+{
+#if DEBUG
```

Comment:
Generally, we replace the function with a no-op define in release mode instead of doing this.